cmd/spire-server/cli/agent/list.go

```diff
@@ -39,8 +48,31 @@ func (listCommand) Synopsis() string {
 
 // Run lists attested agents
 func (c *listCommand) Run(ctx context.Context, env *common_cli.Env, serverClient util.ServerClient) error {
+	filter := &agentv1.ListAgentsRequest_Filter{}
+	if len(c.selectors) > 0 {
+		matchBehavior, err := parseToSelectorMatch(c.matchSelectorsOn)
+		if err != nil {
+			return err
+		}
+
+		selectors := make([]*types.Selector, len(c.selectors))
+		for i, sel := range c.selectors {
+			selector, err := parseSelector(sel)
+			if err != nil {
+				return fmt.Errorf("error parsing selectors: %w", err)
```

Comment:
Similar comment here as from the other pr about convenience of echoing the bad parameter back

```suggestion
				return fmt.Errorf("error parsing selector %q: %w", sel, err)
```